pkg/innerring/processors/container/common.go

```diff
@@ -125,3 +115,37 @@ func (cp *Processor) checkSessionToken(token *session.Token) error {
 
 	return cp.checkKeyOwnership(token, key)
 }
+
+type verbAssert func(*session.ContainerContext) bool
+
+func contextWithVerifiedVerb(tok *session.Token, verbAssert verbAssert) (*session.ContainerContext, error) {
+	c := session.GetContainerContext(tok)
+	if c == nil {
+		return nil, errWrongSessionContext
+	}
+
+	if !verbAssert(c) {
+		return nil, errWrongSessionVerb
+	}
+
+	return c, nil
+}
+
+func checkTokenContext(tok *session.Token, verbAssert verbAssert) error {
+	_, err := contextWithVerifiedVerb(tok, verbAssert)
+	return err
+}
+
+func checkTokenContextWithCID(tok *session.Token, id *cid.ID, verbAssert verbAssert) error {
+	c, err := contextWithVerifiedVerb(tok, verbAssert)
+	if err != nil {
+		return err
+	}
+
+	tokCID := c.Container()
+	if tokCID != nil && !tokCID.Equal(id) {
```

Comment:
Is it valid to have `tokCID == nil`?

Reply:
Yes, it is, here `tokCID == nil` if token is signed with wildcard flag.